server/chat-plugins/helptickets.js

```diff
@@ -758,6 +848,165 @@ const pages = {
+			if (!['staff', 'tickets'].includes(table)) table = 'tickets';
+			let monthString = `${query.shift()}-${query.shift()}`;
+			let date = null;
+			if (monthString.includes('undefined')) {
+				// year/month not provided, use current month
+				date = new Date();
+				monthString = Chat.toTimestamp(date).split(' ')[0].split('-', 2).join('-');
+			}
+			let rawTicketStats = FS(`logs/tickets/${monthString}.csv`).readIfExistsSync();
+			if (!rawTicketStats) return `<div class="pad"><br />No ticket stats found.</div>`;
+
+			// Calculate next/previous month for stats and validate stats exist for the month
+			const prevDate = new Date(monthString);
+			if (!date) date = new Date((prevDate.getMonth() === 11 ? prevDate.getFullYear() + 1 : prevDate.getFullYear()), (prevDate.getMonth() === 11 ? 0 : prevDate.getMonth() + 1));
+			const nextDate = new Date((prevDate.getMonth() >= 10 ? prevDate.getFullYear() + 1 : prevDate.getFullYear()), (prevDate.getMonth() + 2 > 11 ? (prevDate.getMonth() + 2) - 12 : prevDate.getMonth() + 2));
+			let prevString = Chat.toTimestamp(prevDate).split(' ')[0].split('-', 2).join('-');
+			let nextString = Chat.toTimestamp(nextDate).split(' ')[0].split('-', 2).join('-');
+
+			let buttonBar = '';
+			if (FS(`logs/tickets/${prevString}.csv`).readIfExistsSync()) {
+				buttonBar += `<a class="button" href="/view-help-stats-${table}-${prevString}" target="replace" style="float: left">&lt; Previous Month</a>`;
+			} else {
+				buttonBar += `<a class="button disabled" style="float: left">&lt; Previous Month</a>`;
+			}
+			buttonBar += `<a class="button${table === 'tickets' ? ' disabled"' : `" href="/view-help-stats-tickets-${monthString}" target="replace"`}>Ticket Stats</a> <a class="button ${table === 'staff' ? ' disabled"' : `" href="/view-help-stats-staff-${monthString}" target="replace"`}>Staff Stats</a>`;
+			if (FS(`logs/tickets/${nextString}.csv`).readIfExistsSync()) {
+				buttonBar += `<a class="button" href="/view-help-stats-${table}-${nextString}" target="replace" style="float: right">Next Month &gt;</a>`;
+			} else {
+				buttonBar += `<a class="button disabled" style="float: right">Next Month &gt;</a>`;
+			}
+			let col = query.shift() || '';
+
+			let buf = `<div class="pad ladder"><div style="text-align: center">${buttonBar}</div><br />`;
+			buf += `<table style="margin-left: auto; margin-right: auto"><tbody><tr><th colspan="${table === 'tickets' ? 7 : 3}"><h2 style="margin: 5px auto">Help Ticket Stats - ${date.toLocaleString('en-us', {month: 'long', year: 'numeric'})}</h1></th></tr>`;
+			if (table === 'tickets') {
+				if (!['type', 'totaltickets', 'total', 'initwait', 'wait', 'resolution', 'result'].includes(col)) col = 'type';
+				buf += `<tr><th><Button>type</Button></th><th><Button>totaltickets</Button></th><th><Button>total</Button></th><th><Button>initwait</Button></th><th><Button>wait</Button></th><th><Button>resolution</Button></th><th><Button>result</Button></th></tr>`;
```

Comment:
Not really an 'issue' per se, but HTML conventions is to use lowercase for tags (see `<Button>`).

Reply:
This isn't a normal button tag, check down at the end of the page defenition and you will see a replacement regexp that replaces `<Button>text</Button>` with some `<a>` tags.

Reply:
Out of scope for this change, but why?? Why is this not just something we handle with CSS?

Reply:
Reduces the number of lines and amount of various checks I need to get this to work. Its also used for the help ticket form.